datadreamer/pipelines/generate_dataset_from_scratch.py

```diff
@@ -377,8 +371,7 @@ def main():
     os.makedirs(bbox_dir)
 
     # Save arguments
-    with open(os.path.join(save_dir, "generation_args.json"), "w") as f:
-        json.dump(vars(args), f, indent=4)
+    config.save_data(os.path.join(save_dir, "generation_args.yaml"))
```

Comment:
I replaced saving to YAML, which can be reused, instead of saving to JSON.

Reply:
Great idea! Thank you!